peer.go

```diff
@@ -454,6 +424,128 @@ func (p *peer) loadActiveChannels(chans []*channeldb.OpenChannel) error {
 	return nil
 }
 
+// addLink creates and adds a new link from the specified channel.
+func (p *peer) addLink(chanPoint *wire.OutPoint,
+	lnChan *lnwallet.LightningChannel,
+	forwardingPolicy *htlcswitch.ForwardingPolicy,
+	blockEpoch *chainntnfs.BlockEpochEvent,
+	chainEvents *contractcourt.ChainEventSubscription,
+	currentHeight int32, syncStates bool) error {
+
+	linkCfg := htlcswitch.ChannelLinkConfig{
+		Peer:                  p,
+		DecodeHopIterators:    p.server.sphinx.DecodeHopIterators,
+		ExtractErrorEncrypter: p.server.sphinx.ExtractErrorEncrypter,
+		FetchLastChannelUpdate: fetchLastChanUpdate(
+			p.server, p.PubKey(),
+		),
+		DebugHTLC:      cfg.DebugHTLC,
+		HodlMask:       cfg.Hodl.Mask(),
+		Registry:       p.server.invoices,
+		Switch:         p.server.htlcSwitch,
+		Circuits:       p.server.htlcSwitch.CircuitModifier(),
+		ForwardPackets: p.server.htlcSwitch.ForwardPackets,
+		FwrdingPolicy:  *forwardingPolicy,
+		FeeEstimator:   p.server.cc.feeEstimator,
+		BlockEpochs:    blockEpoch,
+		PreimageCache:  p.server.witnessBeacon,
+		ChainEvents:    chainEvents,
+		UpdateContractSignals: func(signals *contractcourt.ContractSignals) error {
+			return p.server.chainArb.UpdateContractSignals(
+				*chanPoint, signals,
+			)
+		},
+		OnChannelFailure: func(chanID lnwire.ChannelID,
```

Comment:
Perhaps move this into it's own method, it would improve the readability of the configuration initialization

Reply:
Could as easily just be raised to a local closure right above the initialization of the config itself.

Reply:
I like the local closure suggestion, as it won't clutter the global surface :)